pkg/registry/scheduling/rest/storage_scheduling.go

```diff
@@ -89,51 +95,99 @@ func (p RESTStorageProvider) v1Storage(apiResourceConfigSource serverstorage.API
+			return false, nil
+		}
 
-			for _, pc := range scheduling.SystemPriorityClasses() {
-				_, err := schedClientSet.PriorityClasses().Get(pc.Name, metav1.GetOptions{})
-				if err != nil {
-					if apierrors.IsNotFound(err) {
-						// TODO: Remove this explicit conversion after scheduling api move to v1
-						v1beta1PriorityClass := &schedulingv1beta1.PriorityClass{}
-						if err := schedulingapiv1beta1.Convert_scheduling_PriorityClass_To_v1beta1_PriorityClass(pc, v1beta1PriorityClass, nil); err != nil {
-							return false, err
-						}
-						_, err := schedClientSet.PriorityClasses().Create(v1beta1PriorityClass)
-						if err != nil && !apierrors.IsAlreadyExists(err) {
-							return false, err
-						} else {
-							klog.Infof("created PriorityClass %s with value %v", pc.Name, pc.Value)
-						}
+		for _, pc := range scheduling.SystemPriorityClasses() {
+			_, err := schedClientSet.PriorityClasses().Get(pc.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// TODO: Remove this explicit conversion after scheduling api move to v1
+					v1beta1PriorityClass := &schedulingv1beta1.PriorityClass{}
+					if err := schedulingapiv1beta1.Convert_scheduling_PriorityClass_To_v1beta1_PriorityClass(pc, v1beta1PriorityClass, nil); err != nil {
+						return false, err
+					}
+					_, err := schedClientSet.PriorityClasses().Create(v1beta1PriorityClass)
+					if err != nil && !apierrors.IsAlreadyExists(err) {
+						return false, err
 					} else {
-						// Unable to get the priority class for reasons other than "not found".
-						klog.Warningf("unable to get PriorityClass %v: %v. Retrying...", pc.Name, err)
-						return false, nil
+						klog.Infof("created PriorityClass %s with value %v", pc.Name, pc.Value)
 					}
+				} else {
+					// Unable to get the priority class for reasons other than "not found".
+					klog.Warningf("unable to get PriorityClass %v: %v. Retrying...", pc.Name, err)
+					return false, nil
 				}
 			}
-			klog.Infof("all system priority classes are created successfully or already exist.")
-			return true, nil
-		})
-		// if we're never able to make it through initialization, kill the API server.
+		}
+		klog.Infof("all system priority classes are created successfully or already exist.")
+		return true, nil
+	})
+}
+
+func AddDefaultSystemQuota(hookContext genericapiserver.PostStartHookContext) error {
+	return wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		client := clientset.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		_, err = client.CoreV1().ResourceQuotas("kube-system").Get("default-system-quota", metav1.GetOptions{})
 		if err != nil {
-			return fmt.Errorf("unable to add default system priority classes: %v", err)
+			if apierrors.IsNotFound(err) {
+				_, err := client.CoreV1().ResourceQuotas("kube-system").Create(systemDefaultQuota())
+				if err != nil && !apierrors.IsAlreadyExists(err) {
+					return false, err
```

Comment:
This looks questionable.  Seems like we should tolerate most errors until the time is up.  Perhaps logging at a lower verbosity or track "lastErr"?

Reply:
Do you want logging to be reduced to something like V(5) or above?

Reply:
I am fine with increasing the log level. Just that, it becomes difficult to debug.